### PR TITLE
 refactor(config): Consolidate logic via serde 

### PR DESCRIPTION
**Pull request by epage**
Configuration is now defined in one place.

Future directions, left off to keep this change small:
- Change merging of args / config to prevent accidentally reading from a
  non-merged value.
- Accept a `--manifest-path` arg and make everything relative to that.
- Support calling `cargo-release` in a child directory.

### CODE REVIEW

**Comment by epage** [comment]
This includes #79.  Unsure if you want to merge the PRs one at a time or just merge this and discard the other.

**Comment by sunng87** [comment]
@epage Fantastic! Using serde to organize config items made it much cleaner. I will get into detail later. 

Also I will probably fix a regression issue metioned #81 before merging this.

Thanks for your effort! Would you like to maintain this library together with me? I can add write permission to you.

**Comment by epage** [comment]
> Also I will probably fix a regression issue metioned #81 before merging this.

Sure, I'll take on resolving when that is merged.  Just let me know when you have a fix in.  In the mean time I'll go work on pulling logic out of `cargo-suity` into `escargot`.

> Thanks for your effort! Would you like to maintain this library together with me? I can add write permission to you.

I'd gladly help maintain.

Another option is I think this would be a good fit for my umbrella org I've been maintaining, [crate-ci](https://github.com/crate-ci/).  The [book](https://crate-ci.github.io/) and examples are the main thing in there right now.  I keep getting distracted from my replacement workflow for trust (cargo-tarball / stager).

**Comment by sunng87** [comment]
@epage just merged #83 and there are something to rebase. 

I wish cargo-release can be a part of the book and am open to any improvement based on your best practices on CI.

**Comment by epage** [comment]
Feel free to contribute a section for the book.  I hadn't done it yet because I hadn't adopted `cargo-release` yet.  My contributing is to make sure I can use it (especially with how annoying workspaces are).

I've gone ahead and given you an invite in case you want to move `cargo-release` over to `crate-ci`.

**Comment by sunng87** [comment]
Thanks @epage . I probably won't move the project at the moment but will check to see if crate-ci a proper fit for this project.

**Comment by epage** [comment]
Ok, got it resolved.  I think I did it correctly.

**Comment by sunng87** [comment]
I have released 0.12.0-beta.1 with these changes included.

File: src/main.rs [hunk 1]
```diff
@@ -168,20 +78,16 @@ fn execute(args: &ReleaseOpt) -> Result<i32, error::FatalError> {
...
-        },
-        release_config.as_ref(),
-        config::ENABLE_FEATURES,
-    );
-    let all_features = get_bool_option(
-        args.all_features,
-        release_config.as_ref(),
-        config::ENABLE_ALL_FEATURES,
-    );
+        } else if release_config.enable_features.is_empty() {
```
Comment (sunng87) [inline src/main.rs]: this should be `! release_config.enable_features.is_empty()`

I have fixed on master branch.

Comment (epage) [reply src/main.rs]: Thanks!

**Pull request merged**